app/medInria/medRegistrationWorkspace.cpp

```diff
@@ -130,123 +111,75 @@ bool medRegistrationWorkspace::isUsable()
-    if(!movingView)
-    {
-        qWarning() << "Non layered view are not suported yet in registration workspace.";
-        return;
-    }
-
-    medAbstractData *movingData = movingView->layerData(movingView->currentLayer());
-
-    medAbstractLayeredView* fuseView  = dynamic_cast<medAbstractLayeredView*>(d->fuseContainer->view());
-    if(fuseView)
-        fuseView->removeData(movingData);
-
-    d->fuseContainer->addData(movingData);
-    fuseView  = dynamic_cast<medAbstractLayeredView*>(d->fuseContainer->view());
-
-    if(movingData)
-    {
-        d->viewGroup->addImpactedView(movingView);
-        d->viewGroup->addImpactedView(fuseView);
+    updateFromContainers(0);
+}
 
-        d->movingLayerGroup->addImpactedlayer(movingView, movingData);
-        d->movingLayerGroup->addImpactedlayer(fuseView, movingData);
-    }
-    if (!d->selectorToolBox->setMovingData(movingData))
-    {
-        // delete the view because something failed at some point
-        d->viewGroup->removeImpactedView(movingView);
-        d->movingLayerGroup->removeImpactedlayer(movingView, movingData);
-        d->movingLayerGroup->removeImpactedlayer(fuseView, movingData);
-        movingView->deleteLater();
-    }
+void medRegistrationWorkspace::updateFromMovingContainer()
+{
+    updateFromContainers(1);
 }
 
-void medRegistrationWorkspace::updateFromFixedContainer()
+void medRegistrationWorkspace::updateFromContainers(int containerIndex)
 {
-    if(!d->selectorToolBox)
-        return;
+    medRegistrationSelectorToolBox* toolbox = dynamic_cast<medRegistrationSelectorToolBox*>(selectorToolBox());
 
-    if(!d->fixedContainer->view())
+    if(toolbox)
     {
-        medAbstractLayeredView* fuseView  = dynamic_cast<medAbstractLayeredView*>(d->fuseContainer->view());
-        if(fuseView)
+        if(!d->containers[containerIndex]->view())
         {
-            if(fuseView->layer(d->selectorToolBox->fixedData()) == 0)
+            medAbstractLayeredView* fuseView  = dynamic_cast<medAbstractLayeredView*>(d->containers[2]->view());
+            if(fuseView)
             {
-                d->fuseContainer->removeView();
-                d->fuseContainer->addData(d->selectorToolBox->movingData());
+                medAbstractData* dataInView      = containerIndex==0? toolbox->fixedData() : toolbox->movingData();
+                medAbstractData* dataInOtherView = containerIndex==0? toolbox->movingData() : toolbox->fixedData();
+
+                if(fuseView->layer(dataInView) == 0)
+                {
+                    d->containers[2]->removeView();
+                    d->containers[2]->addData(dataInOtherView);
+                }
+                else
+                {
+                    fuseView->removeLayer(1);
+                }
             }
-            else
-                fuseView->removeLayer(1);
 
+            containerIndex==0? toolbox->setFixedData(NULL) : toolbox->setMovingData(NULL);
```

Comment:
It's very uncommon (i.e. I've never seen it :)) to use the conditional operator this way, in place of an if-else statement. It's usually used to assign a conditional value to a variable.